### ISSUE TITLE
Windows, MSYS2/MINGW64, mingw-w64-x86_64-boost 1.75.0-2: build cbd0b69 error

### ISSUE THREAD

**Issue by pashamcr** [facebook/redex#596]
```
[ 65%] Building CXX object CMakeFiles/redex.dir/opt/remove-unused-args/RemoveUnusedArgs.cpp.obj
In file included from C:/msys64/mingw64/include/boost/range/concepts.hpp:24,
                 from C:/msys64/mingw64/include/boost/range/detail/any_iterator_wrapper.hpp:16,
                 from C:/msys64/mingw64/include/boost/range/detail/any_iterator.hpp:22,
                 from C:/msys64/mingw64/include/boost/range/any_range.hpp:17,
                 from C:/msys64/home/pasha/src/redex/libredex/IRInstruction.h:10,
                 from C:/msys64/home/pasha/src/redex/libredex/IRCode.h:18,
                 from C:/msys64/home/pasha/src/redex/libredex/CallGraph.h:13,
                 from C:/msys64/home/pasha/src/redex/service/type-analysis/GlobalTypeAnalyzer.h:10,
                 from C:/msys64/home/pasha/src/redex/opt/type-analysis/GlobalTypeAnalysisPass.h:10,
                 from C:/msys64/home/pasha/src/redex/opt/remove-unreachable/TypeAnalysisAwareRemoveUnreachable.h:10,
                 from C:/msys64/home/pasha/src/redex/opt/remove-unreachable/TypeAnalysisAwareRemoveUnreachable.cpp:8:
C:/msys64/mingw64/include/boost/range/value_type.hpp: In instantiation of 'struct boost::range_value<long long unsigned int>':
C:/msys64/mingw64/include/boost/algorithm/string/join.hpp:46:9:   required by substitution of 'template<class SequenceSequenceT, class Range1T> typename boost::range_value<T>::type boost::algorithm::join(const SequenceSequenceT&, const Range1T&) [with SequenceSequenceT = long long unsigned int; Range1T = <missing>]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeSet.h:581:29:   required from 'boost::intrusive_ptr<sparta::pt_impl::PatriciaTree<IntegerType> > sparta::pt_impl::insert(IntegerType, const boost::intrusive_ptr<sparta::pt_impl::PatriciaTree<IntegerType> >&) [with IntegerType = long long unsigned int]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeSet.h:210:42:   required from 'sparta::PatriciaTreeSet<Element>& sparta::PatriciaTreeSet<Element>::insert(Element) [with Element = const DexType*]'
C:/msys64/home/pasha/src/redex/libredex/DexTypeEnvironment.h:161:24:   required from here
C:/msys64/mingw64/include/boost/range/value_type.hpp:26:12: error: no type named 'type' in 'struct boost::range_iterator<long long unsigned int, void>'
   26 |     struct range_value : iterator_value< typename range_iterator<T>::type >
      |            ^~~~~~~~~~~
C:/msys64/mingw64/include/boost/range/value_type.hpp: In instantiation of 'struct boost::range_value<unsigned int>':
C:/msys64/mingw64/include/boost/algorithm/string/join.hpp:46:9:   required by substitution of 'template<class SequenceSequenceT, class Range1T> typename boost::range_value<T>::type boost::algorithm::join(const SequenceSequenceT&, const Range1T&) [with SequenceSequenceT = unsigned int; Range1T = sparta::ptmae_impl::MapValue<unsigned int, sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)> >::ValueInterface]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMap.h:758:36:   required from 'boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> > sparta::ptmap_impl::update(sparta::ptmap_impl::CombiningFunction<typename Value::type>&, IntegerType, const typename Value::type&, const boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> >&) [with IntegerType = unsigned int; Value = sparta::ptmae_impl::MapValue<unsigned int, sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)> >::ValueInterface; sparta::ptmap_impl::CombiningFunction<typename Value::type> = std::function<sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>(const sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>&, const sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>&)>; typename Value::type = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>; typename Value::type = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMap.h:319:52:   required from 'sparta::PatriciaTreeMap<Key, ValueType, Value>& sparta::PatriciaTreeMap<Key, ValueType, Value>::insert_or_assign(Key, const mapped_type&) [with Key = unsigned int; ValueType = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>; Value = sparta::ptmae_impl::MapValue<unsigned int, sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)> >::ValueInterface; sparta::PatriciaTreeMap<Key, ValueType, Value>::mapped_type = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMapAbstractEnvironment.h:273:27:   required from 'void sparta::ptmae_impl::MapValue<Variable, Domain>::insert_binding(const Variable&, const Domain&) [with Variable = unsigned int; Domain = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMapAbstractEnvironment.h:108:38:   required from 'sparta::PatriciaTreeMapAbstractEnvironment<Variable, Domain>& sparta::PatriciaTreeMapAbstractEnvironment<Variable, Domain>::set(const Variable&, const Domain&) [with Variable = unsigned int; Domain = sparta::FiniteAbstractDomain<Nullness, sparta::BitVectorLattice<Nullness, 5, std::hash<int> >, std::bitset<5>, (& lattice)>]'
C:/msys64/home/pasha/src/redex/libredex/NullnessDomain.h:149:55:   required from here
C:/msys64/mingw64/include/boost/range/value_type.hpp:26:12: error: no type named 'type' in 'struct boost::range_iterator<unsigned int, void>'
C:/msys64/mingw64/include/boost/range/value_type.hpp: In instantiation of 'struct boost::range_value<short unsigned int>':
C:/msys64/mingw64/include/boost/algorithm/string/join.hpp:46:9:   required by substitution of 'template<class SequenceSequenceT, class Range1T> typename boost::range_value<T>::type boost::algorithm::join(const SequenceSequenceT&, const Range1T&) [with SequenceSequenceT = short unsigned int; Range1T = sparta::ptmae_impl::MapValue<short unsigned int, DexTypeDomain>::ValueInterface]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMap.h:758:36:   required from 'boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> > sparta::ptmap_impl::update(sparta::ptmap_impl::CombiningFunction<typename Value::type>&, IntegerType, const typename Value::type&, const boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> >&) [with IntegerType = short unsigned int; Value = sparta::ptmae_impl::MapValue<short unsigned int, DexTypeDomain>::ValueInterface; sparta::ptmap_impl::CombiningFunction<typename Value::type> = std::function<DexTypeDomain(const DexTypeDomain&, const DexTypeDomain&)>; typename Value::type = DexTypeDomain; typename Value::type = DexTypeDomain]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMap.h:872:18:   required from 'boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> > sparta::ptmap_impl::merge(sparta::ptmap_impl::CombiningFunction<typename Value::type>&, const boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> >&, const boost::intrusive_ptr<sparta::ptmap_impl::PatriciaTree<IntegerType, Value> >&) [with IntegerType = short unsigned int; Value = sparta::ptmae_impl::MapValue<short unsigned int, DexTypeDomain>::ValueInterface; sparta::ptmap_impl::CombiningFunction<typename Value::type> = std::function<DexTypeDomain(const DexTypeDomain&, const DexTypeDomain&)>; typename Value::type = DexTypeDomain]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMap.h:327:46:   required from 'sparta::PatriciaTreeMap<Key, ValueType, Value>& sparta::PatriciaTreeMap<Key, ValueType, Value>::union_with(const combining_function&, const sparta::PatriciaTreeMap<Key, ValueType, Value>&) [with Key = short unsigned int; ValueType = DexTypeDomain; Value = sparta::ptmae_impl::MapValue<short unsigned int, DexTypeDomain>::ValueInterface; sparta::PatriciaTreeMap<Key, ValueType, Value>::combining_function = std::function<DexTypeDomain(const DexTypeDomain&, const DexTypeDomain&)>]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMapAbstractEnvironment.h:293:23:   required from 'sparta::AbstractValueKind sparta::ptmae_impl::MapValue<Variable, Domain>::meet_like_operation(const sparta::ptmae_impl::MapValue<Variable, Domain>&, std::function<Domain(const Domain&, const Domain&)>) [with Variable = short unsigned int; Domain = DexTypeDomain]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMapAbstractEnvironment.h:260:12:   required from 'sparta::AbstractValueKind sparta::ptmae_impl::MapValue<Variable, Domain>::meet_with(const sparta::ptmae_impl::MapValue<Variable, Domain>&) [with Variable = short unsigned int; Domain = DexTypeDomain]'
C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeMapAbstractEnvironment.h:259:21:   required from here
C:/msys64/mingw64/include/boost/range/value_type.hpp:26:12: error: no type named 'type' in 'struct boost::range_iterator<short unsigned int, void>'
make[2]: *** [CMakeFiles/redex.dir/build.make:2539: CMakeFiles/redex.dir/opt/remove-unreachable/TypeAnalysisAwareRemoveUnreachable.cpp.obj] Ошибка 1
make[2]: *** Ожидание завершения заданий…
In file included from C:/msys64/home/pasha/src/redex/sparta/include/PatriciaTreeSetAbstractDomain.h:15,
                 from C:/msys64/home/pasha/src/redex/service/dataflow/Liveness.h:12,
                 from C:/msys64/home/pasha/src/redex/opt/remove-unused-args/RemoveUnusedArgs.cpp:19:
C:/msys64/home/pasha/src/redex/sparta/include/PowersetAbstractDomain.h: In member function 'bool sparta::PowersetAbstractDomain<Element, Powerset, Snapshot, Derived>::contains(const Element&) const [with Element = unsigned int; Powerset = sparta::ptsad_impl::SetValue<unsigned int>; Snapshot = const sparta::PatriciaTreeSet<unsigned int>&; Derived = sparta::PatriciaTreeSetAbstractDomain<unsigned int>]':
C:/msys64/home/pasha/src/redex/sparta/include/PowersetAbstractDomain.h:183:3: warning: control reaches end of non-void function [-Wreturn-type]
  183 |   }
      |   ^
make[1]: *** [CMakeFiles/Makefile2:212: CMakeFiles/redex.dir/all] Ошибка 2
make: *** [Makefile:171: all] Ошибка 2
```

**Comment by pashamcr** [comment]
Sorry for thinking aloud, but finding a bump in a bare field and tripping over it ... you need to be able to... 10 o'clock - not a single answer, is the star that gathered you a star? Once again I apologize, I did not sleep well. I don't look inside the problem, I know it is ... why don't you fix it.
(sorry again, fucking google translator)

**Comment by agampe** [comment]
Should be fixed (by 2367bc63f75022e0ce10caf24a303ce3615a154f).
